app/src/org/commcare/views/notifications/NotificationMessageFactory.java

```diff
@@ -211,7 +227,18 @@ public static NotificationMessage message(MessageTag message, String[] parameter
                 //No big deal, key doesn't need to exist
             }
 
-            return new NotificationMessage(customCategory, title, detail, action, new Date());
+            NotificationActionButtonInfo buttonInfo = null;
+            try {
+                String button = parameters[3] == null ? Localization.get(base + ".button") : Localization.get(base + ".button", new String[]{parameters[3]});
```

Comment:
I don't think we need to support a parameterised localisation for button text. We can simply keep `parameters` as an array of [3] as before and eval button as just `Localization.get(base + ".button")`